midpoint/install/system-requirements/index.adoc

```diff
@@ -223,93 +230,123 @@ Refer to the xref:/midpoint/release/[midPoint Releases] documentation for softwa
+
+Primarily, the nodes communicate with the central Quartz library rather than among themselves,
+with the exception of *cache invalidation*.
+When a node changes data in the midPoint database, the node informs other nodes about the need to invalidate their cache.
+The *communication between nodes runs over HTTPS*.
+
+==== Node Management Access
+
+Identity engineers need to access all nodes, either using the web UI or the REST API, be it via a load balancer or directly. 
+Depending on the setup, they may need to connect to the nodes through a VPN which tunnels them to a particular node or the load balancer.
+
+==== All Nodes Are Created Equal
+
+Similarly, each midPoint node needs full access to all the resources other nodes have.
+These are, for instance, the shared HA DB repository, the source and target systems, or an SMTP server for notifications.
+All nodes need to have the same access level regardless of whether you deploy midPoint on 200 nodes or just one.
+This is because *all nodes are created equal*: when one node goes down, others need to replace it in full.
+
+You can check that connections to resource work as expected using `ping`, `telnet`, or `wget`, for instance.
 
 == Environment Requirements
 
-In IAM projects, at least two environments are typically used: test & production.
-In many cases, there is also a local midPoint installation on the identity engineer's workstation, and another development environment in the customer's infrastructure.
+Usually, at least two environments are typically used for the development of an IAM deployment: test and production.
+In many cases, there is also a local midPoint installation on the identity engineer's workstation and a separate development environment in the customer's infrastructure.
+
+=== Keep the Environments as Similar as Possible
 
-The best practice is to use a configuration that is as similar as possible in all of these environments.
-However, it should also be completely isolated without access, for example, from the test midPoint environment to production Application 1.
+The best practice is to use a configuration that is as similar as possible in all these environments.
+However, the environments should also be completely isolated so that the test environment cannot touch production data on a resource.
 VPN can be shared.
 
-For identity development, it is recommended for the test and production environments to have the same operating system (and version), application version, and data for all source and target systems.
-Any differences may lead to situations where something is working and is well tested in one environment, but does not work in another environment.
+We recommend having the same operating system, midPoint version, and resource data for all environments.
+If _same_ is not possible, as similar as possible is desirable.
+Any differences may lead to situations where something is working and is well tested in one environment, but does not work in another.
```

Comment:
We can add note about simulations and be able to safely verify new configuration on production instance before it's deployed for real.